src/warc2zim/content_rewriting/generic.py

```diff
@@ -0,0 +1,84 @@
+from urllib.parse import urlsplit
+
+from jinja2.environment import Template
+from warcio.recordloader import ArcWarcRecord
+
+from warc2zim.content_rewriting.css import CssRewriter
+from warc2zim.content_rewriting.ds import build_domain_specific_rewriter
+from warc2zim.content_rewriting.html import HtmlRewriter
+from warc2zim.url_rewriting import ArticleUrlRewriter
+from warc2zim.utils import get_record_content, get_record_mime_type, get_record_url
+
+
+class Rewriter:
+    def __init__(
+        self,
+        path: str,
+        record: ArcWarcRecord,
+        known_urls: set[str],
+    ):
+        self.content = get_record_content(record)
+
+        mimetype = get_record_mime_type(record)
+
+        self.path = path
+        self.orig_url_str = get_record_url(record)
+        self.url_rewriter = ArticleUrlRewriter(self.orig_url_str, known_urls)
+
+        self.rewrite_mode = self.get_rewrite_mode(record, mimetype)
+
+    def rewrite(
+        self, head_template: Template, css_insert: str | None
+    ) -> tuple[str, str | bytes]:
+        opts = {}
+
+        if self.rewrite_mode == "html":
+            return self.rewrite_html(head_template, css_insert)
+
+        if self.rewrite_mode == "css":
+            return self.rewrite_css()
+
+        if self.rewrite_mode == "javascript":
+            return self.rewrite_js(opts)
+
+        return ("", self.content)
+
+    def get_rewrite_mode(self, record, mimetype):
+        if getattr(record, "method", "GET") == "POST":
+            return None
+        if mimetype == "text/html":
+            # TODO : Handle header "Accept" == "application/json"
```

Comment:
What does it mean? Do you plan to do it soon? Is there an open issue around what has to be done?

Reply:
Wabac is a checking if the header `Accept` is `application/json` and then use the json converter even if mimetype is `text/html`.

We don't have a json rewriter (we probably need one) and so I've just left a comment here to keep a trace that we may have something to do here.

> Do you plan to do it soon?

This has to be discussed. Until now, we had implement transformation based on what needed (this is why we agreed to not keep response header for now). But we may change our strategy and reimplement (blindly) the transformation in wabac. I will open a issue.

Reply:
I've open an issue to track the question (at least): https://github.com/openzim/warc2zim/issues/160